src/ToolUpdate/UpdateCalculator.php

```diff
@@ -124,4 +131,21 @@ private function calculateTasksToExecute(RepositoryInterface $desired, array $to
 
         return $tasks;
     }
+
+    private function describeUpgradeDirection(
```

Comment:
Should probably be named `getTaskType()` as it is only for `upgrade` an `UpgradeDirection` and even no direction at all for `reinstall`.

otoh., I'd prefer to have this method return the entire action string.